feat: add `stats/incr/nanme`

satansin123 opened the pull request:
Resolves #5570 .

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds the @stdlib/stats/incr/nanme package, which computes an incremental mean error while handling NaN values by skipping them.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5570 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

![image](https://github.com/user-attachments/assets/1b74f126-8d66-4e77-af37-6b949ca4fa10)
![image](https://github.com/user-attachments/assets/de4e5357-f8c4-4916-98f4-6b2fbd348a1d)
![image](https://github.com/user-attachments/assets/1f80fabf-d3d7-4918-b58c-281010ad6b68)



## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

satansin123 committed:
feat(stats): add nanme package

satansin123 committed:
refractor(stats/examples): simplify conditional logic

satansin123 committed:
refractor: update package description and example code

satansin123 committed:
docs: improve documentation in README

satansin123 committed:
Merge branch 'develop' of https://github.com/stdlib-js/stdlib into feature/adding-nanme-package-5570
Merge upstream develop into feature/adding-nanme-package-5570

satansin123 committed:
style: fix code formatting in README example

satansin123 commented:
/stdlib lint-autofix

stdlib-bot committed:
fix: resolve lint errors

satansin123 commented:
/stdlib help

satansin123 commented:
/stdlib check-files

satansin123 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

satansin123 committed:
refractor: correct lint error

satansin123 committed:
refractor: correct test lint error

satansin123 commented:
/stdlib lint-autofix

satansin123 committed:
refractor: correct test lint error

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanme/lib/main.js:
```diff
@@ -0,0 +1,78 @@
+
+// MODULES //
+
+var incrmean = require( '@stdlib/stats/incr/mean' );
```
This should wrap `@stdlib/stats/incr/me` directly rather than reimplementing the mean error logic via `incrmean`. The standard `nan*` wrapper pattern delegates to the corresponding base accumulator.

```suggestion
var incrme = require( '@stdlib/stats/incr/me' );
```

Then on line 56 use `var me = incrme();` and on line 71 use `return me( x, y );`.

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanme/examples/index.js:
```diff
@@ -0,0 +1,46 @@
+		v2 = ( randu()*100.0 ) - 50.0;
+	}
+	err = accumulator( v1, v2 );
+	console.log( '%d\t%d\t%d', v1.toFixed( 3 ), v2.toFixed( 3 ), ( err === null ) ? NaN : err.toFixed( 3 ) );
```
`.toFixed()` returns a string. Use `%s`, not `%d`.

Planeshifter reviewed (changes requested):
Thanks for the PR!

Planeshifter committed:
fix: address review feedback

- Refactor to wrap incrme instead of reimplementing with incrmean
- Fix variable name inconsistency in lib/index.js
- Fix format specifiers in examples/index.js
- Add missing accumulator call in README example loop
- Use relative require path in test.js and benchmark.js
- Update copyright years to 2026

Planeshifter commented:
/stdlib merge

Planeshifter reviewed (approved):
Let's land it!
